tests/components/websocket_api/test_commands.py

```diff
@@ -3480,3 +3481,281 @@ async def test_extract_from_target_validation_error(
     assert msg["type"] == const.TYPE_RESULT
     assert not msg["success"]
     assert "error" in msg
+
+
+@patch("annotatedyaml.loader.load_yaml")
+@patch.object(Integration, "has_triggers", return_value=True)
+async def test_get_triggers_for_target(
+    mock_has_triggers: Mock,
+    mock_load_yaml: Mock,
+    hass: HomeAssistant,
+    websocket_client: MockHAClientWebSocket,
+    area_registry: ar.AreaRegistry,
+    device_registry: dr.DeviceRegistry,
+    entity_registry: er.EntityRegistry,
+    label_registry: lr.LabelRegistry,
+) -> None:
+    """Test get_triggers_for_target command with mixed target types."""
+
+    async def async_get_triggers(hass: HomeAssistant) -> dict[str, type]:
+        return {
+            "turned_on": Mock,
+            "non_target_trigger": Mock,
+        }
+
+    mock_platform(hass, "light.trigger", Mock(async_get_triggers=async_get_triggers))
+    mock_platform(hass, "switch.trigger", Mock(async_get_triggers=async_get_triggers))
+    mock_platform(hass, "sensor.trigger", Mock(async_get_triggers=async_get_triggers))
+    mock_platform(
+        hass,
+        "mqtt.trigger",
+        Mock(
+            async_get_triggers=AsyncMock(
+                return_value={
+                    "_": Mock,
+                    "light_message": Mock,
+                    "light_flash": Mock,
+                }
+            )
+        ),
+    )
```

Comment:
Instead of mocking real domains, I'd suggest to mock test domains, maybe `platform_1` and so on?
The reason is that you create test lights which are not lights etc., it's just confusing IMO.

Reply:
Renamed `mqtt` to `component1`.
I want to keep the base domains (light, switch, sensor), because those are "special" and match all entities of that platform independently of the integration domain.